### ISSUE TITLE
Questions about the transformer

### ISSUE THREAD

**Issue by xmy0916** [ninatu/everything_at_once#3]
Hello, thanks for your work! I have a question about the transformer, i see that all the modalities forward by one transformer structure. Have you tested using different transformer for different modality? Put another way, Is there any benefit to using only one transformer?

thx~

**Comment by ninatu** [comment]
Hi! We used one transformer to enable fusion between modalities, but in our ablation, we also considered using different transformers for different modalities (Table 4 “single-mod. transformer per mod”). The config for this experiment is [here](https://github.com/ninatu/everything_at_once/blob/main/configs/pretraining/ablation/architecture/singe_mod_tr_per_mod.yaml). For t->va retrieval, we found it beneficial to use one transformer. With different transformers for different modalities, video and audio modalities are processed separately in corresponding transformers, and then video and audio embeddings are just summed up to obtain a video-audio representation. However, when we use one transformer (which is trained on all modality combinations) we forward tokens from both video and audio modalities together, allowing them to attend to each other, and therefore we obtain a fused video-audio representation.
